### ISSUE TITLE
Flow Not Saving Selected System TAG

### ISSUE THREAD

**Issue by skykingisepic** [nextcloud/workflow_pdf_converter#436]
## Steps to reproduce
1. Add new flow when Tag Assigned and File System Tag is tagged with
2. Select System tag 'pdf'
3. Keep orig, overwrite existing

### Expected behaviour
Active flow shows selected tag

### Actual behaviour
Selected Tag disappears, not saved so no pdf created when assigning tag to doc

## PDF Converter app

**PDF Converter app version:** (see apps admin page: `/index.php/settings/apps`)
1.10.0

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:**

**Nextcloud Version:** (see admin page)
25.02

**Where did you install Nextcloud from:**

**Signing status:**

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```

**List of activated apps:**

<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**

<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

### Server log (data/nextcloud.log)
<details>
```
Insert your server log here
```
</details>

**Comment by ikonnow** [comment]
I am having the same issue with the workflow PDF converter in the newest version of Nextcloud. The PDF files are not being generated and the result is no action. Has there been any progress in finding a solution for this?

**Comment by rotdrop** [comment]
I have an issue which is similar. Actually, I can configure the workflow to be triggered the tag, but then looking at lines

https://github.com/nextcloud/workflow_pdf_converter/blob/814ccb6739913900085bf3b29d0ae4f290bb9c42/lib/Operation.php#L79-L82

we see that only events of type `GenericEvent` AKA `OCP\EventDispatcher\GenericEvent` are handled.

Now, inserting some "printf"-style debugging lines we see that the event which is actually triggered is of class `OCP\SystemTag\MapperEvent` which extends directly `OCP\EventDispatcher\Event` and hence is not a `GenericEvent`.

The `MapperEvent` also does not directly contain the file-system node but instead -- if the event is related to a file, as determined by `getObjectType()` -- is able to return the file-id which then would have to be converted to an FS node.

So in short: PDF conversion can at the moment not be triggered by system tags. To necessary changes are straight forward but more involved than just adding a semicolon or two ;)

**Comment by rotdrop** [comment]
Actually, this is also sort of a bug as it is not clear that PDF conversion cannot (yet) be triggered by changing collaborative tags.

**Comment by rotdrop** [comment]
BTW, part of this issue, namely that the tags were not remembered, is also discussed here nextcloud/server#3564  and probably fixed by 163b86b729fc6cf4105fa7ad95e488e6c5c05532

Still: even with that changer PDF conversion on tag changed cannot work unless something alike #454 is added to this app.

**Comment by ostasevych** [comment]
Still not working with tag assigning :( Are any working fixes available?
